modules/julia/CMakeLists.txt

```diff
@@ -0,0 +1,130 @@
+include(${CMAKE_CURRENT_LIST_DIR}/FindJulia.cmake)
+
+if(NOT Julia_FOUND)
+    ocv_module_disable(julia)
+elseif(NOT PYTHON3_EXECUTABLE)
+#    message(WARNING "Python 3 required for Julia bindings...disabling")
+#    ocv_module_disable(julia)
+endif()
+
+math(EXPR ARCH "${CMAKE_SIZEOF_VOID_P} * 8")
+if (${ARCH} EQUAL 32 AND ${Julia_WORD_SIZE} MATCHES "64")
+    warn_mixed_precision("32" "64")
+    ocv_module_disable(julia)
+    return()
+elseif (${ARCH} EQUAL 64 AND NOT ${Julia_WORD_SIZE} MATCHES "64")
+    warn_mixed_precision("64" "32")
+    ocv_module_disable(julia)
+    return()
+endif()
+
+if(NOT JlCxx_DIR)
+execute_process(
+    COMMAND "${Julia_EXECUTABLE}" --startup-file=no -e "using CxxWrap; print(CxxWrap.CxxWrapCore.prefix_path())"
+    OUTPUT_VARIABLE JlCxx_DIR_prefix
+)
+
+if(JlCxx_DIR_prefix)
+    set(JlCxx_DIR ${JlCxx_DIR_prefix})
```

Comment:
you should check that JlCxx_Config.cmake is inside ${JlCxx_DIR}. If not, try the macOS variant (lib/cmake, as in the patch that I suggested)